crypto/buffer/buffer.c

```diff
@@ -141,7 +145,6 @@ int BUF_MEM_grow_clean(BUF_MEM *str, size_t len)
         return (len);
     }
     if (str->max >= len) {
-        memset(&str->data[str->length], 0, len - str->length);
```

Comment:
I'm not sure this is a good idea. Assuming you've always used the BUF_MEM_grow_clean() function (and no other bugs writing too much), that data should be all 0s. But BUF_MEM_new() doesn't do this for instance. I don't think you can rely on it already being 0 while the function promises to turn it into 0.

The same goes for the other change in this function, you try to optimize it by starting at the previous max instead of the previous length.

In short I think this whole function should be left as it was.